api/apiclient.go

```diff
@@ -829,14 +829,9 @@ func (d dialer) dial(_ <-chan struct{}) (io.Closer, error) {
 
 // dial1 makes a single dial attempt.
 func (d dialer) dial1() (jsoncodec.JSONConn, *tls.Config, error) {
-	tlsConfig := utils.SecureTLSConfig()
+	tlsConfig := NewTLSConfig(d.opts.certPool)
 	tlsConfig.InsecureSkipVerify = d.opts.InsecureSkipVerify
-	if d.opts.certPool != nil {
-		// We want to be specific here (rather than just using "anything").
-		// See commit 7fc118f015d8480dfad7831788e4b8c0432205e8 (PR 899).
-		tlsConfig.RootCAs = d.opts.certPool
-		tlsConfig.ServerName = "juju-apiserver"
-	} else {
+	if d.opts.certPool == nil {
```

Comment:
could we do all this in NewTLSConfig() below

Reply:
the cert pool nil check? no, because it's specific to this code path